recipe/activate.sh

```diff
@@ -1,10 +1,26 @@
 #!/bin/bash
-if [ "$(uname)" == "Darwin" ]
-then
-    # for Mac OSX
-    export KERAS_BACKEND=tensorflow
-elif [ "$(uname)" == "Linux" ]
-then
-    # for Linux
-    export KERAS_BACKEND=theano
+
+# get the location of the keras.json file
+KERAS_JSON="$HOME/.keras/keras.json"
+
+if [ -f $KERAS_JSON ]; then # if the file exists
+
+  BACKEND_KEY="backend" # value we are interested in finding
+  KEY_VALUE_REGEX="\"($BACKEND_KEY)\": \"([^\"]*)\"" # regex
+  JSON_TEXT=$(cat $KERAS_JSON) # data in keras.json file
+
+  if [[ $JSON_TEXT =~ $KEY_VALUE_REGEX ]]; then # check for a match
+    BACKEND_VALUE="${BASH_REMATCH[2]}" # get the user defined backend value
+    export KERAS_BACKEND=$BACKEND_VALUE # set the environmental variable
+  fi
```

Comment:
@jakirkham Yes, perhaps I missed that in the Keras documentation, but from experience and just hacking around I empirically determined that was the behavior.

I'm submitting this PR because 1) it still maintains the current functionality of setting the backend environment based on the OS but 2) reads the contents of the user's `keras.json` file and respects their choices.

As I've said several times now, I don't think that there is anything wrong with having the conda recipe ship with the `keras_activation.sh` script and set defaults - one of the best things about using the conda  recipes is that someone else has done the work of setting good defaults so it "just works" when you install it. However, the current setup leads to unclear behavior because the `keras_activation.sh` script sets the `KERAS_BACKEND` variable without notifying the user, and overrides the preferences set in the `keras.json` file. 

So my suggested fixes for this are either 1) accept the PR which will allow the backend to be set by the `keras.json` file, as is the expected behavior based on the Keras documentation or 2) modify the installation/activation script such that the user is warned that the `keras_activation.sh` script will be setting the backend for them, and reasonable ways they can change that behavior (eg showing them the location of the script, ways to delete or modify etc).